_overviews/scala3-book/domain-modeling-oop.md

```diff
@@ -173,6 +173,61 @@ trait ComposedService extends GreetingService, TranslationService
 Abstract members in one trait (such as `translate` in `GreetingService`) are automatically matched with concrete members in another trait.
 This not only works with methods as in this example, but also with all the other abstract members mentioned above (that is, types, value definitions, etc.).
 
+### Mixing traits in on the fly
+
+Traits that have concrete methods can be mixed into classes on the fly. Given a class:
+
+{% tabs traits_6 %}
+{% tab 'Scala 2 and 3' %}
+```scala
+class MyService(name: String)
+```
+{% endtab %}
+{% endtabs %}
+
+you can create a `MyService` instance that mixes in the traits when you create it:
+
+{% tabs traits_7 %}
+{% tab 'Scala 2 and 3' %}
+```scala
+val s = new MyService("ComposedService") with GreetingService with TranslationService
```

Comment:
Intuitively, after seeing the syntax used above, someone could expect this to be defined in Scala 3 as:
```
val s = MyService("ComposedService") with GreetingService, TranslationService
```
It seems like only the Scala 2 syntax is accepted, though.